db/db-design-notes.md

```diff
@@ -0,0 +1,193 @@
+# Database Design Notes
+
+-------------------------------------------------------------------------------
+## 2021-08-12 (Robyn)
+
+
+### Journeys table
+- I have removed the complex journey name column and just put in origin and 
+  destination columns. I don't think we should bother that it's technically not 
+  3NF as it makes our lives much easier. The origin and destination can just be
+  stored directly in the db when a user saves their journey.
+
+
+### Carbon saved / emissions table
+- Column added for emissions saved.
+- The plan is that a user will enter origin, destination and vehicle. The total 
+  emissions for the journey with the specified vehicle is returned alongside a 
+  list of alternative vehicle options and their associated emissions. The user 
+  tells the app if they want to save their journey and which vehicle they ended 
+  up using. It is the data of the journey made and vehicle selected as the one 
+  they used that gets stored in the DB. The app also calculates the difference 
+  between their originally chosen vehicle and the vehicle they end up using and 
+  **stores this in the emissions saved column** for that journey.
+
```

Comment:
So the function that we end up building for offset/saving needs to take into consideration emissions based on initial user choice and then emissions based on final user choice. The difference should result in the emissions saved.
I had initially envisaged it as user input to just be location and destination, modes of transport proposed by our service and offset calculated against the max emissions that could possibly be generated by a mode of transport in our proposed function. Of course the proposed mode of transport should ideally  never going to take into consideration any vehicle not listed in the user vehicle table.
Not sure if this makes sense?

Reply:
Yes, I understand. This is just what I understood from our meeting the other day. We can do how you described here. Maybe we should have another proper discussion about it with everyone before we write the function.